cli/pkg/kctrl/cmd/app/init/vendir.go

```diff
@@ -0,0 +1,228 @@
+			return err
+		}
+	case FetchChartFromGit:
+		gitStep := NewGitStep(v.ui, v.config)
+		err := Run(gitStep)
+		if err != nil {
+			return err
+		}
+	}
+	includedPaths, err := v.getIncludedPaths()
+	if err != nil {
+		return err
+	}
+	v.config.Directories[0].Contents[0].IncludePaths = includedPaths
+	return SaveVendir(v.config)
+	return nil
+}
+
+func (v *VendirStep) initializeVendirDirectorySection() []vendirconf.Directory {
+	var directory vendirconf.Directory
+	directory = vendirconf.Directory{
+		Path: VendirSyncDirectory,
+		Contents: []vendirconf.DirectoryContents{
+			{
+				Path: ".",
+			},
+		},
+	}
+	directories := []vendirconf.Directory{directory}
+	v.config.Directories = directories
+	SaveVendir(v.config)
+	return directories
+}
+
+func (v *VendirStep) getIncludedPaths() ([]string, error) {
+	v.ui.PrintInformationalText("We need exact manifest files from the above provided repository which should be included. Multiple files can be included using a comma separator. If you want to include all the files, enter *.")
+	includedPaths := v.config.Directories[0].Contents[0].IncludePaths
+	defaultIncludedPath := strings.Join(includedPaths, ",")
+	if len(includedPaths) == 0 {
+		defaultIncludedPath = IncludeAllFiles
+	}
+	textOpts := ui.TextOpts{
+		Label:        "Enter the paths which need to be included as part of this package",
+		Default:      defaultIncludedPath,
+		ValidateFunc: nil,
+	}
+	path, err := v.ui.AskForText(textOpts)
+	path = strings.TrimSpace(path)
+	if err != nil {
+		return nil, err
+	}
+	if path == IncludeAllFiles {
+		return []string{}, nil
+	}
+	paths := strings.Split(path, ",")
+	return paths, nil
+}
+
+func (v *VendirStep) PostInteract() error {
+	v.ui.PrintInformationalText("We will use vendir to fetch the data from source to local directory. Vendir allows to declaratively state what should be in a directory and sync data sources into it. All the information entered above has been persisted into vendir.yml file.")
+	err := v.printVendirFile()
+	if err != nil {
+		return err
+	}
+	err = v.runVendirSync()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (v *VendirStep) printVendirFile() error {
+	vendirFileLocation := VendirFileName
+	v.ui.PrintActionableText(fmt.Sprintf("Printing %s", vendirFileLocation))
+	v.ui.PrintCmdExecutionText(fmt.Sprintf("cat %s", vendirFileLocation))
+	err := v.printFile(vendirFileLocation)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (v *VendirStep) printFile(filePath string) error {
+	var stdoutBs, stderrBs bytes.Buffer
+
+	localCmdRunner := exec.NewPlainCmdRunner()
+	cmd := goexec.Command("cat", []string{filePath}...)
+	cmd.Stdin = nil
+	cmd.Stdout = &stdoutBs
+	cmd.Stderr = &stderrBs
+	localCmdRunner.Run(cmd)
+
+	result := exec.CmdRunResult{
+		Stdout: stdoutBs.String(),
+		Stderr: stderrBs.String(),
+	}
+	if result.Error != nil {
+		return fmt.Errorf("Printing file: %s", result.Stderr)
+	}
+
+	v.ui.PrintCmdExecutionOutput(result.Stdout)
+
+	return nil
+}
```

Comment:
lets not run commands to print vendir.yml. if we want to show it, then lets just use regular read file functions. cat may not be available in all environments (e.g. windows)

Reply:
By doing this, I think that we are slowly moving towards abstraction rather than being transparent about what is happening under the hood.

Reply:
Adopted